web/src/EducationBenchmarking.Web/Controllers/SchoolPlanningStepsController.cs

```diff
@@ -196,12 +196,79 @@ public async Task<IActionResult> PrePopulateData(string urn, int year, [FromForm
 
     [HttpGet]
     [Route("timetable")]
-    public IActionResult Timetable(string urn, int year)
+    public async Task<IActionResult> Timetable(string urn, int year)
     {
         using (logger.BeginScope(new { urn, year }))
         {
             try
             {
+                var school = await establishmentApi.GetSchool(urn).GetResultOrThrow<School>();
+                var plan = await benchmarkApi.GetFinancialPlan(urn, year).GetResultOrThrow<FinancialPlan>();
+
+                if (plan.UseFigures == null)
+                {
+                    throw new ArgumentNullException(nameof(plan.UseFigures));
+                }
+
+                var backAction = plan.UseFigures == true
+                    ? Url.Action("PrePopulateData", new { urn, year })
+                    : Url.Action("TotalNumberTeachers", new { urn, year });
+
+                ViewData[ViewDataConstants.Backlink] = new BacklinkInfo(backAction);
+
+                var viewModel = new SchoolPlanViewModel(school, year, plan);
+
+                return View(viewModel);
+            }
+            catch (Exception e)
+            {
+                logger.LogError(e, "An error displaying school curriculum and financial planning: {DisplayUrl}",
+                    Request.GetDisplayUrl());
+                return e is StatusCodeException s ? StatusCode((int)s.Status) : StatusCode(500);
+            }
+        }
+    }
+
+    [HttpPost]
+    [Route("timetable")]
+    public async Task<IActionResult> Timetable(string urn, int year, int? timetable)
+    {
+        using (logger.BeginScope(new { urn, year, }))
+        {
+            try
+            {
+                var school = await establishmentApi.GetSchool(urn).GetResultOrThrow<School>();
+                var plan = await benchmarkApi.GetFinancialPlan(urn, year).GetResultOrThrow<FinancialPlan>();
+                plan.Timetable = timetable;
+
+                if (plan.UseFigures == null) 
+                {
+                    throw new ArgumentNullException(nameof(plan.UseFigures));
+                }
+
+                var backAction = plan.UseFigures == true
+                    ? Url.Action("PrePopulateData", new { urn, year })
+                    : Url.Action("TotalNumberTeachers", new { urn, year });
+
+                ViewData[ViewDataConstants.Backlink] = new BacklinkInfo(backAction);
+
+                if (!timetable.HasValue)
```

Comment:
This can be simplified.

```
                if (timetable is null or <= 0)
                {
                    var msg = !timetable.HasValue ? "Please enter a value" : "Value must be a whole number";
                    var viewModel = new SchoolPlanViewModel(school, year, plan);
                    ModelState.AddModelError(nameof(SchoolPlanViewModel.Timetable),msg );
                    return View(viewModel);
                }
```

Reply:
Fixed in latest commit